frontend/src/components/Admin/TeamEvent/TeamEventDashboard.tsx

```diff
@@ -0,0 +1,77 @@
+import React, { useState, useEffect } from 'react';
+import { Table, Header, Loader } from 'semantic-ui-react';
+import { useMembers } from '../../Common/FirestoreDataProvider';
+import { TeamEventsAPI } from '../../../API/TeamEventsAPI';
+import { REQUIRED_LEAD_TEC_CREDITS, REQUIRED_MEMBER_TEC_CREDITS } from '../../../consts';
+import styles from './TeamEventDashboard.module.css';
+
+const calculateMemberCreditsForEvent = (member: IdolMember, event: TeamEvent): number =>
+  event.attendees.reduce((val: number, attendee) => {
+    if (attendee.member.email !== member.email) {
+      return val;
+    }
+    if (event.hasHours && attendee.hoursAttended)
```

Comment:
This is OK, but I wonder if it makes it sense for this calculation to go into a `TeamEventsUtils` file. Aira's got something really similar here: https://github.com/cornell-dti/idol/pull/437.

Reply:
Good point but let's deal with that in separate PR after both of these are done.